plugin/evm/vm.go

```diff
@@ -1378,6 +1383,33 @@ func (vm *VM) getBlock(_ context.Context, id ids.ID) (snowman.Block, error) {
 	return vm.newBlock(ethBlock)
 }
 
+// GetAcceptedBlock attempts to retrieve block [blkID] from the VM. This method
+// only returns accepted blocks.
+func (vm *VM) GetAcceptedBlock(ctx context.Context, blkID ids.ID) (snowman.Block, error) {
+	blk, err := vm.GetBlock(ctx, blkID)
+	if err != nil {
+		return nil, err
+	}
+
+	lastAcceptedBlock := vm.LastAcceptedBlock()
+	height := blk.Height()
+	if lastAcceptedBlock.Height() < height {
+		// The provided block is not decided.
+		return nil, database.ErrNotFound
+	}
+
+	acceptedBlkID, err := vm.GetBlockIDAtHeight(ctx, height)
```

Comment:
Should we move the last accepted height check into `GetBlockIDAtHeight`? The implementation in coreth does not follow the avalanchego specification for this method.